### DISCUSSION TITLE
Schema内部的状态如何与外部状态交互进行更新

### DISCUSSION THREAD

**Discussion by dxiongya** [alibaba/formily#3799] [Q&A 2.x]
我假设有一个schema: 
{'x-component': "Input", 'x-component-props': {icon: loading ? 'loading' : null}}

上面的loading状态是一个接口的请求，我这样去写是无效的，有什么方案能够让外部状态和schema内部进行交互？

**Comment by xiaosi-ss** [comment]
setComponentProps and  loading以scope的形式传进去

**Reply by dxiongya** [reply]
试过scope没有变，setComponentProps是可以，想问一下scope这个怎么传递有效？

**Comment by janryWang** [comment]
可以看看 https://formilyjs.org/zh-CN/guide/advanced/controlled 这篇文档

**Reply by dxiongya** [reply]
嗯嗯，感谢回复，我已经知道了~
